src/openssl/ssl/socket.cr

```diff
@@ -95,6 +95,10 @@ abstract class OpenSSL::SSL::Socket < IO
     LibSSL.ssl_set_bio(@ssl, @bio, @bio)
   end
 
+  def io
```

Comment:
Maybe there could be a delegate to `io.remote_address` instead, so we don't care whether `io` is a Socket or SSLsocket.

Reply:
I think it makes sense that it is exposed here, since we're "unwrapping" the SSLSocket.

If you were to delegate `remote_address` to SSLSocket there would likely be duplication in handling various sockets.

Reply:
I don't understand your claim. We musn't care if it's a SSLSocket just like we don't care whether is a UnixSocket or TCPSocket or whatever other IO. If `io` responds to `#remote_address` then it returns an address. Nothing more.